vision/standard_object/odlc_classify_shape.py

```diff
@@ -69,420 +89,339 @@ def process_shapes(
-    vec_b = np.squeeze(vec_b)
-
-    return np.degrees(
-        np.arccos(np.dot(vec_a, vec_b) / (np.linalg.norm(vec_a) * np.linalg.norm(vec_b)))
-    )
-
-
-def get_angles(approx: consts.Contour) -> npt.NDArray[npt.Shape["*"], npt.Float64]:
+    x_avg: Float64 = 0
+    y_avg: Float64 = 0
+    numPoints: int = 0
+    point: NDArray[Shape["2, 2"], IntC, Float64]
+    # Finds average x and y value
+    for point in cnt:
+        y_avg += point[0][0]
+        x_avg += point[0][1]
+        numPoints += 1
+
+    x_avg = x_avg // numPoints
+    y_avg = y_avg // numPoints
+    i: int = 0
+    # Centers the shape at (0,0) to allow for a better scan of the shape in polar coordinates
+    for point in cnt:
+        point[0][0] -= y_avg
+        point[0][1] -= x_avg
+        i += 1
+
+    pol_cnt: NDArray[Shape["*, 2"], Float64] = cartesian_array_to_polar(
+        cnt
+    )  # Converts array of rectangular coordinates (x,y) to polar (angle, radius)
+    pol_cnt_sorted: NDArray[Shape["*, 2"], Float64] = merge_sort(pol_cnt)
+    x: NDArray[Shape["*"], Float64]
+    y: NDArray[Shape["*"], Float64]
+    x, y = condense_polar(pol_cnt_sorted)
+
+    return x, y
+
+
+def condense_polar(
+    polar_array: NDArray[Shape["*, 2"], Float64]
+) -> NDArray[Shape["128,2"], Float64]:
     """
-    Will find the angle at every point in the given approximated contour.
+    Condenses a polar array to have 'NUM_STEPS' data points for analysis
 
     Parameters
     ----------
-    approx : consts.Contour
-        Approximated version of the contour (polygon approximation with cv2.approxPolyDP())
+    polar_array : List[float]
+        A polar array of unknown length
 
     Returns
     -------
-    angles : npt.NDArray[npt.Shape["*"], npt.Float64]
-        1d array of angles for each corresponding vertex in the given approximated contour
-    """
-    angles: npt.NDArray[npt.Shape["*"], npt.Float64] = np.empty(approx.shape[0], npt.Float64)
-    idx: int
-    a_pt: npt.NDArray[npt.Shape["1, 2"], npt.IntC]  # one point of the approx contour
-    for idx, a_pt in enumerate(approx):
-        angles[idx] = get_angle((approx[idx - 1], a_pt, approx[(idx + 1) % approx.shape[0]]))
+    newx, newy : List[float], List[float]
 
-    return angles
-
-
-def compare_angles(
-    angles: npt.NDArray[npt.Shape["*"], npt.Float64], compare_angle: float, thresh: float
-) -> npt.NDArray[npt.Shape["*"], npt.Bool8]:
     """
-    Takes an array, angles, and compares each element against a specific angle.
+    # Converting data to a form able to be passed into scipy.interp1d
+    x: NDArray[Shape["*"]] = np.empty(len(polar_array))
+    y: NDArray[Shape["*"]] = np.empty(len(polar_array))
+    for i in range(len(polar_array)):
+        x[i] = polar_array[i][1]
+        y[i] = polar_array[i][0]
+
+    # Linear interpolation to normalize all shapes to have the same number of data points
+    newx: NDArray[Shape["128"], Float64] = np.linspace(x.min(), x.max(), num=NUM_STEPS)
```

Comment:
Since you are using NEW_STEPS to determine how many steps are in the array, you should not use 128 in your typehints for NDArrays.